### ISSUE TITLE
Mark All Read fails on notifications page

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/Panoptes-Front-End#5433]
## Expected behavior
Mark All Read should mark my unread notifications as read.

## Current behavior
_Please include any error messages from the browser console and/or screenshots_
The button sometimes fails to work. When it does fail, reloading the page does not fix it.
The error is `Uncaught TypeError: Cannot read property 'update' of undefined` and it seems to be happening inside a `forEach` loop, according to the stack trace.

## Steps to replicate
It's an intermittent bug, that only seems to be triggered by certain notifications. When it does happen, it can be triggered by pressing Mark All Read on a section in the notifications page.

## Additional information
This seems to be the loop that's failing.
https://github.com/zooniverse/Panoptes-Front-End/blob/cedef380766d05dde2151e09ba367ba5d8fbaee1/app/pages/notifications/notification-section.jsx#L131-L133

- **Operating system:** OSX Mojave
- **Browser:** Chrome

**Comment by lcjohnso** [comment]
A bump here to say this is actively happening to me now -- behavior I'm seeing seems identical to bug described. Could be related to a deleted mention on Zooniverse Talk that triggered the error state (see screenshot below)?
![Screen Shot 2019-07-15 at 10 04 27 PM](https://user-images.githubusercontent.com/2006101/61262926-bb1b6a80-a74c-11e9-9893-d4900efab75b.png)

**Comment by lcjohnso** [comment]
Bumping this here to prevent from going stale: this is still occuring (I still have one unread DM I cannot read or delete).  One possible source of this issue: deleting of spam messages without reading them first.  Spam messages are removed from view, but not actually deleted, leading to this error state.

**Comment by srallen** [comment]
To be honest, this seems like more a back end issue than front end. I don't think we have a good way to deal with deleting spam and my guess is that when those messages were "deleted", their read state wasn't changed from unread to read at the same time of deletion. I think the Talk API needs an administrative method to correctly mark unread messages as deleted then read in a batch or something like that.

**Comment by camallen** [comment]
yes I agree - these messages need to be cleaned up properly in the talk api. 

One other point is that talk doesn't have a notion of read messages :( see https://github.com/zooniverse/Talk-Api/issues/160

**Comment by srallen** [comment]
That's unfortunate. I made an assumption that it did.

**Comment by eatyourgreens** [comment]
Bumping this because I'm seeing it again for my Talk notifications. The JS error is `unable to read property 'update' of undefined` during a `forEach` loop.

![Screenshot of Talk notifications showing the console error when Mark All Read fails.](https://user-images.githubusercontent.com/59547/88526648-189af600-cff4-11ea-8aa2-0ca2df978475.png)

**Comment by eatyourgreens** [comment]
An existence check for `data.notification` should fix this, right?
https://github.com/zooniverse/Panoptes-Front-End/blob/cedef380766d05dde2151e09ba367ba5d8fbaee1/app/pages/notifications/notification-section.jsx#L131-L133

**Comment by goplayoutside3** [comment]
@eatyourgreens @camallen There's a new report on Talk of this notifications bug, and I wanted to bring it to your attention to further document - https://www.zooniverse.org/talk/17/2409220

**Comment by eatyourgreens** [comment]
@goplayoutside3 Thanks! I've also been having this same problem on my account recently. After reading back through the conversation here, I think it's down to the Talk API not having a concept of 'read messages'.

**Comment by goplayoutside3** [comment]
This bug was recently reported via a Freshdesk [ticket](https://zooniverse.freshdesk.com/a/tickets/23183) where an unread notification could not be cleared due to the original comment being deleted before the user's notification could be marked 'read'. It was solved by a backend dev manually finding the row in the talk db/notifications table and flipping the DELIVERED bit to TRUE.
